logic/src/androidMain/kotlin/com/wire/kalium/logic/CoreLogic.kt

```diff
@@ -19,15 +25,22 @@ import kotlinx.coroutines.runBlocking
  * and it should only override functions/variables from CoreLogicCommon
  */
 actual class CoreLogic(
-    private val applicationContext: Context,
+    private val appContext: Context,
     clientLabel: String,
     rootProteusDirectoryPath: String
 ) : CoreLogicCommon(clientLabel, rootProteusDirectoryPath) {
 
+    override fun getSessionRepo(): SessionRepository {
+        val sessionPreferences = KaliumPreferencesSettings(EncryptedSettingsHolder(appContext, SHARED_PREFERENCE_FILE_NAME).encryptedSettings)
+        val sessionStorage: SessionStorage = SessionStorageImpl(sessionPreferences)
+        val sessionLocalRepository: SessionLocalRepository = SessionLocalDataSource(sessionStorage, sessionMapper)
+        return SessionDataSource(sessionLocalRepository)
+    }
+
     override fun getAuthenticationScope(): AuthenticationScope =
         AuthenticationScope(
```

Comment:
Shouldn't you inject the `sessionRepository` here as well now that it's a singleton.

Reply:
Nice catch, I missed this one